packages/core/src/child-proxy/child-process-proxy.ts

```diff
@@ -51,11 +52,23 @@ export class ChildProcessProxy<T> implements Disposable {
     fileDescriptions: FileDescriptions,
     pluginModulePaths: readonly string[],
     workingDirectory: string,
-    execArgv: string[]
+    execArgv: string[],
+    idGenerator: IdGenerator
   ) {
-    this.worker = childProcess.fork(fileURLToPath(new URL('./child-process-proxy-worker.js', import.meta.url)), { silent: true, execArgv });
+    const workerId = idGenerator.next().toString();
+    this.worker = childProcess.fork(fileURLToPath(new URL('./child-process-proxy-worker.js', import.meta.url)), {
+      silent: true,
+      execArgv,
+      env: { STRYKER_MUTATOR_WORKER: workerId },
```

Comment:
```suggestion
      env: { STRYKER_MUTATOR_WORKER: workerId,  ...process.env },
```

We shouldn't forget to also pass along the `process.env` of the parent process.